x-pack/plugins/security_solution/public/common/components/url_state/helpers.ts

```diff
@@ -170,23 +238,25 @@ export const makeMapStateToProps = () => {
         {}
       );
 
-    return {
-      urlState: {
-        ...searchAttr,
-        [CONSTANTS.sourcerer]: selectedPatterns,
-        [CONSTANTS.timerange]: {
-          global: {
-            [CONSTANTS.timerange]: globalTimerange,
-            linkTo: globalLinkTo,
-          },
-          timeline: {
-            [CONSTANTS.timerange]: timelineTimerange,
-            linkTo: timelineLinkTo,
-          },
+    const urlState: UrlState = {
+      ...searchAttr,
+      [CONSTANTS.sourcerer]: selectedPatterns,
+      [CONSTANTS.timerange]: {
+        global: {
+          [CONSTANTS.timerange]: globalTimerange,
+          linkTo: globalLinkTo,
+        },
+        timeline: {
+          [CONSTANTS.timerange]: timelineTimerange,
+          linkTo: timelineLinkTo,
         },
-        [CONSTANTS.timeline]: timeline,
       },
+      [CONSTANTS.timeline]: timeline,
     };
+
+    if (detailPanel != null) urlState[CONSTANTS.detailPanel] = detailPanel;
```

Comment:
nit: consider avoiding mutating `urlState` after initialization by moving this check:

```tsx
if (detailPanel != null) urlState[CONSTANTS.detailPanel] = detailPanel;
``` 

to the initialization of `urlState`, e.g.:

```tsx
    const urlState: UrlState = {
      // ...
      [CONSTANTS.timeline]: timeline,
      ...(detailPanel != null ? { [CONSTANTS.detailPanel]: detailPanel } : undefined),
    };
```